### DISCUSSION TITLE
Docker data uses more space than the actual library

### DISCUSSION THREAD

**Discussion by carterlando** [immich-app/immich#11032] [Q&A]
### The bug

I've installed the docker version of Immich in order to add 2 external libraries.
One is 333 Gb and the other one is 383 Gb (around 700Gb in total), stored on a separate 1TB HDD.
The docker root is on a separate 1TB SSD.
The upload directory is on the same 1TB SSD.

The docker files keep increasing and finally reached the limit of the 1TB SSD, with even more thumbnails that need to be generated.
Instead of number of thumbnails generated to go down as they are processed, it keeps increasing.

This is my second attempt at importing the external library as initially the docker root was set on the same 128GB as the Ubuntu installation. But that attempt was short lived as 128 GB were quickly filled by the docker data and got corrupted.

That's when I pointed the docker root to the larger SSD but with the same result. Now the docker data is larger than the actual external library that's on another HDD.

Restarting the containers doesn't clear the data.

**Comment by carterlando** [comment]
After using
`docker compose down -v`
And then
`docker system prune`
The space is not reclaimed.

I reclaimed the space with `docker volume prune`

This time I plan to load a single external library and turn on Immich jobs one by one to see which one is causing the problem for me.

**Comment by carterlando** [comment]
I imported the libraries one by one and ran the jobs one by one as well.
It eventually hanged at generating thumbnails for one of the libraries but after a docker restart it continued just fine.
Now the docker data is just above 80Gb and I can continue Immich as intended.

**Comment by carterlando** [comment]
I have to reopen this issue because the docker volume it's using again all the available disk space where it shouldn't need more than 80 Gb.

![Screenshot 2024-07-11 163146](https://github.com/immich-app/immich/assets/27959710/3512d232-7447-40fe-a7fe-d12af943bda3)

It looks that the volume is for the Redis container.

![Screenshot 2024-07-11 163355](https://github.com/immich-app/immich/assets/27959710/5f34f7c7-9499-4dc9-b9b5-c3f68862f383)

Is there a way to clean this data and prevent it from growing?

**Comment by bo0tzz** [comment]
That's Redis doing something weird so it's not strictly an Immich issue. You can probably just set whatever flags Redis needs to disable persistence.

**Comment by carterlando** [comment]
Thanks for your input @bo0tzz 
I ended up removing the volume and updating the docker-compose.yml to this:

  redis:
    container_name: immich_redis
    image: registry.hub.docker.com/library/redis:6.2-alpine@sha256:84882e87b54734154586e5f8abd4dce69fe7311315e2fc6d67c29614c8de2672
    restart: always
    _command: redis-server --appendonly no --save ""_
    
 No idea what I'm doing, I just hope it works :)
